src/Cohesity.Powershell/Scripts/StorageDomain/Set-CohesityStorageDomain.ps1

```diff
@@ -24,13 +62,17 @@ function Set-CohesityStorageDomain {
 
         $token = $session.Accesstoken.Accesstoken
     }
+
     Process {
         $payload = $null
 
         if ($StorageDomain -ne $null) {
             $payload = $StorageDomain
             if ($Name) {
                 $payload = $StorageDomain | Where-Object {$_.name -eq $Name}
+            } elseif ($NewDomainName) {
+                Write-Warning 'Cannot update entire storage domain with same domain name. Please specify any single storage domain to update'
```

Comment:
Unsure what this means

Reply:
If the user is using the pipeline and not providing which domain to be updated, then we will update all the available domains.
In case if the user wants to update the name of the domain and not provided which domain's name to be updated, then we can't update all the domains with the same name. So, have added this validation in case of updating the name.